flink-libraries/flink-state-processing-api/src/test/java/org/apache/flink/state/api/input/MultiStateKeyIteratorTest.java

```diff
@@ -125,4 +226,117 @@ public void testIteratorRemovesFromAllDescriptors() throws Exception {
                             .count());
         }
     }
+
+    /** Test for lazy enumeration of inner iterators. */
+    @Test
+    public void testIteratorPullsSingleKeyFromAllDescriptors() throws AssertionError {
```

Comment:
IIUC, you want to use this case to verify the key numer you iterator is correct ?
So you should iterator until it doesn't hasNext, right ?

Reply:
Okay, Thanks for the clarification.